crypto/crypto_test.go

```diff
@@ -28,69 +28,94 @@ import (
 )
 
 func TestGeneratePrivateKey(t *testing.T) {
+	ecdsaAlgos := []crypto.SignatureAlgorithm{
+		crypto.ECDSA_P256,
+		crypto.ECDSA_secp256k1,
+	}
+
+	nonECDSAAlgos := []crypto.SignatureAlgorithm{
+		crypto.BLS_BLS12381,
+	}
```

Comment:
Also minor: I think from the SDK point of view, it expects the same behavior from non-supported algorithms or non-valid algorithms, so I guess `crypto.SignatureAlgorithm(-42)` could even be included in this array. 

It's just a comment to point this out, but I let you decide how to implement this test.

Reply:
I intentionally treated this as a separate case because `-42` does not represent a valid algorithm in any sense, whereas BLS is valid but _not supported_ by the SDK. I updated the test to be more clear about this distinction.

Reply:
I see your point 👌